### DISCUSSION TITLE
How to avoid 403 Forbidden errors with fetch_channel()?

### DISCUSSION THREAD

**Discussion by lexicalunit** [Rapptz/discord.py#9814] [Q&A] [answered]
I've got a piece of code that accepts a `discord.Guild` and a Channel ID (an `int` value) and I want to know if my bot should be allowed to do execute `client.fetch_channel(channel_id)` without encountering a permissions issue. What I've tried to do is just:
```python
async def foo(client: discord.Client, guild: discord.Guild, channel_id: int):
    if not guild.me.guild_permissions.read_messages:
        return None
    return client.fetch_channel(channel_id)
```
But every now and then, I do get the occasional 403 Forbidden error from this code:
```log
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/spellbot/operations.py", line 153, in safe_fetch_text_channel
    fetched = await retry(lambda: client.fetch_channel(channel_xid))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/ddtrace/internal/compat.py", line 163, in func_wrapper
    result = await coro(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/spellbot/operations.py", line 47, in retry
    return await func()
           ^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/discord/client.py", line 2565, in fetch_channel
    data = await self.http.get_channel(channel_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/discord/http.py", line 739, in request
    raise Forbidden(response, data)
discord.errors.Forbidden: 403 Forbidden (error code: 50001): Missing Access
```
Just wondering if I can maybe prevent this from happening by making a different permissions check? AFAIK I should only need `read_messages` permissions to fetch a channel tho 🤔

**Comment by Rapptz** [comment, marked as answer]
You're checking for guild permissions but channel permissions can forbid someone from viewing a channel.

**Reply by lexicalunit** [reply]
Ah, right. Tho to find out if that is the case I'd have to first fetch the channel, and then do `fetched_channel.permissions_for(guild.me).read_messages`... but that's not possible because even the fetching fails. I guess just a hole in the Discord API 😢
